mempool/src/interface/mock.rs

```diff
@@ -0,0 +1,143 @@
+        Self::new()
+    }
+}
+
+impl MempoolInterfaceMock {
+    pub fn new() -> MempoolInterfaceMock {
+        MempoolInterfaceMock {
+            add_transaction_called: Arc::new(Mutex::new(false)),
+            add_transaction_should_error: Arc::new(Mutex::new(false)),
+            get_all_called: Arc::new(Mutex::new(false)),
+            get_all_should_error: Arc::new(Mutex::new(false)),
+            contains_transaction_called: Arc::new(Mutex::new(false)),
+            contains_transaction_should_error: Arc::new(Mutex::new(false)),
+            collect_txs_called: Arc::new(Mutex::new(false)),
+            collect_txs_should_error: Arc::new(Mutex::new(false)),
+            subscribe_to_events_called: Arc::new(Mutex::new(false)),
+            subscribe_to_events_should_error: Arc::new(Mutex::new(false)),
+            run_called: Arc::new(Mutex::new(false)),
+            run_should_error: Arc::new(Mutex::new(false)),
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl MempoolInterface for MempoolInterfaceMock {
+    async fn add_transaction(&mut self, _tx: SignedTransaction) -> Result<(), Error> {
+        *self.add_transaction_called.lock().unwrap() = true;
+
+        if *self.add_transaction_should_error.lock().unwrap() {
+            Err(Error::SubsystemFailure)
+        } else {
+            Ok(())
+        }
+    }
+
+    async fn get_all(&self) -> Result<Vec<SignedTransaction>, Error> {
+        *self.get_all_called.lock().unwrap() = true;
+
+        if *self.get_all_should_error.lock().unwrap() {
+            Err(Error::SubsystemFailure)
+        } else {
+            Ok(vec![])
+        }
+    }
+
+    async fn contains_transaction(&self, _tx: &Id<Transaction>) -> Result<bool, Error> {
+        *self.contains_transaction_called.lock().unwrap() = true;
+
+        if *self.contains_transaction_should_error.lock().unwrap() {
+            Err(Error::SubsystemFailure)
+        } else {
+            Ok(true)
+        }
+    }
+
+    async fn collect_txs(
+        &self,
+        tx_accumulator: Box<dyn TransactionAccumulator + Send>,
+    ) -> Result<Box<dyn TransactionAccumulator>, Error> {
+        *self.collect_txs_called.lock().unwrap() = true;
+
+        if *self.collect_txs_should_error.lock().unwrap() {
+            Err(Error::SubsystemFailure)
+        } else {
+            Ok(tx_accumulator)
+        }
+    }
+
+    async fn subscribe_to_events(
+        &mut self,
+        _handler: Arc<dyn Fn(MempoolEvent) + Send + Sync>,
+    ) -> Result<(), Error> {
+        *self.subscribe_to_events_called.lock().unwrap() = true;
+
+        if *self.subscribe_to_events_should_error.lock().unwrap() {
+            Err(Error::SubsystemFailure)
+        } else {
+            Ok(())
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl MempoolSubsystemInterface for MempoolInterfaceMock {
+    async fn run(
+        mut self,
+        mut call_rq: CallRequest<dyn MempoolInterface>,
+        _shut_rq: ShutdownRequest,
```

Comment:
Why is shutdown request not handled. Is that intentional?

Reply:
Not intentional, I just didn't need it at the time and missed it during cleanup. I'll fix this, thanks.